r/R/trajectory_analysis.R

```diff
@@ -0,0 +1,68 @@
+runTrajectoryAnalysis <- function(req, data) {
+  root_nodes <- req$body$rootNodes
+
+  cell_data <- generateGraphData(data)
+  cell_data <- monocle3::order_cells(cell_data, reduction_method = "UMAP", root_pr_nodes = root_nodes)
+
+  pseudotime <- as.data.frame(cell_data@principal_graph_aux@listData$UMAP$pseudotime)
+  pseudotime$cells_id <- data@meta.data$cells_id
+  pseudotime <- pseudotime[order(pseudotime$cells_id), ]
+  pseudotime <- pseudotime %>%
+    tidyr::complete(cells_id = seq(0, max(data@meta.data$cells_id))) %>%
+    dplyr::select(-cells_id)
+
+  return(unname(pseudotime))
+}
+
+runGenerateTrajectoryGraph <- function(req, data) {
```

Comment:
why does this function take a req argument? it does not seem to be used

Reply:
it was not necessary. Removed